src/System.Private.CoreLib/shared/System/Environment.Unix.cs

```diff
@@ -366,12 +367,12 @@ public static unsafe string UserName
             get
             {
                 // First try with a buffer that should suffice for 99% of cases.
-                string username;
+                string? username;
                 const int BufLen = Interop.Sys.Passwd.InitialBufferSize;
                 byte* stackBuf = stackalloc byte[BufLen];
                 if (TryGetUserNameFromPasswd(stackBuf, BufLen, out username))
```

Comment:
This can return true and set userName to null... it only returns false when the buffer was too small.

Reply:
Good catch.  Fixed.